src/utils/auth_utils.js

```diff
@@ -77,77 +84,71 @@ const addRoleToCookies = async cookies => {
 const refreshToken = async () => {
   const currentUser = await getCurrentUser(auth);
   if (currentUser) {
-    const refreshUrl = `https://securetoken.googleapis.com/v1/token?key=${process.env.REACT_APP_FIREBASE_APIKEY}`;
     const refreshT = currentUser.refreshToken;
-    try {
-      const {
-        data: { access_token: idToken },
-      } = await axios.post(refreshUrl, {
-        grant_type: 'refresh_token',
-        refresh_token: refreshT,
-      });
-      // Sets the appropriate cookies after refreshing access token
-      setCookie(cookieKeys.ACCESS_TOKEN, idToken, cookieConfig);
-      const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
-      setCookie(cookieKeys.ROLE, user.data.user.role, cookieConfig);
-      return idToken;
-    } catch (e) {
-      return null;
-    }
+    const {
+      data: { access_token: idToken },
+    } = await axios.post(refreshUrl, {
+      grant_type: 'refresh_token',
+      refresh_token: refreshT,
+    });
+    // Sets the appropriate cookies after refreshing access token
+    setCookie(cookieKeys.ACCESS_TOKEN, idToken, cookieConfig);
+    const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
+    setCookie(cookieKeys.ROLE, user.data.user.role, cookieConfig);
+    return idToken;
   }
   return null;
 };
 
-// This response interceptor will refresh the user's access token using the refreshToken helper method
-NPOBackend.interceptors.response.use(
-  response => {
-    return response;
-  },
-  async error => {
-    if (error.response) {
-      const { status, data } = error.response;
-      switch (status) {
-        case 400:
-          // check if 400 error was token
-          if (data === '@verifyToken no access token') {
-            // token has expired;
-            try {
-              // attempting to refresh token;
-              await refreshToken();
-              // token refreshed, reattempting request;
-              const { config } = error.response;
-              // configure new request in a new instance;
-              return await axios({
-                method: config.method,
-                url: `${config.baseURL}${config.url}`,
-                data: config.data,
-                params: config.params,
-                headers: config.headers,
-                withCredentials: true,
-              });
-            } catch (e) {
-              /**
-               * This returns the next action to perform if there's an error with refreshing the token
-               */
-              return window.location.replace('/');
+const createAxiosResponseInterceptor = axiosInstance => {
```

Comment:
Rename this function to make it clear it adds an interceptor handling authentication, something like `addAuthInterceptor`. Also this function isn't called anywhere, so the interceptor never gets added. Call it after the function definition with `addAuthInterceptor(NPOBackend);`

Reply:
Yeah it probably shouldn't be called here. Could you move `NPOBackend` to a generic utils file, then call this function there? Would bring our implementation a bit closer to the current NPO project structure.

Reply:
Makes sense